blockmanager.go

```diff
@@ -2228,7 +2181,8 @@ func (b *blockManager) BlockHeadersSynced() bool {
 
 	// If our time source (median times of all the connected peers) is at
 	// least 24 hours ahead of our best known block, we aren't current.
-	minus24Hours := b.cfg.TimeSource.AdjustedTime().Add(-24 * time.Hour)
+	minus24Hours := b.cfg.blkHdrProcessor.timeSource.AdjustedTime().Add(
```

Comment:
See general comment about exposting the functionality as method directly, so just `blkHdrProcessor. AdjustedTime()` instead of `blkHdrProcessor.timeSource.AdjustedTime()`. You can likely achieve this via struct embedding.